lib/commands/local/flash.ts

```diff
@@ -75,6 +75,21 @@ export default class LocalFlashCmd extends Command {
 			LocalFlashCmd,
 		);
 
+		if (process.platform === 'linux') {
+			const { promisify } = await import('util');
+			const { exec } = await import('child_process');
+			const execAsync = promisify(exec);
+			const info = await execAsync('cat /proc/version');
```

Comment:
I don't know if it would be possible for this statement to throw an error in some Linux distribution... Perhaps it would be a bit more robust to have something along the lines of:
```ts
		let distroVersion = '';
		try {
			const info = await execAsync('cat /proc/version');
			distroVersion = info.stdout.toLowerCase();
		} catch {
			// pass
		}
		if (distroVersion.includes('microsoft')) {
			throw new ExpectedError(stripIndent`
				This command is known not to work on WSL. Please use a CLI release
				for Windows (not WSL), or balenaEtcher.`);
		}
```
Above I've also replaced `process.exit(0)` with `throw new ExpectedError`, which allows "graceful exit" through `errors.ts`. This also changes the process exit code from 0 to 1, which I think is good too.

Reply:
> graceful exit

`process.exit()` causes immediate process termination, preventing network requests from completing (e.g. HTTP to Amplitude or Sentry) and I believe preventing buffered filesystem I/O from being flushed to disk.

Reply:
https://nodejs.org/docs/latest-v10.x/api/process.html#process_process_exit_code
>Calling process.exit() will force the process to exit as quickly as possible even if there are still asynchronous operations pending that have not yet completed fully, including I/O operations to process.stdout and process.stderr.
> In most situations, it is not actually necessary to call process.exit() explicitly. The Node.js process will exit on its own if there is no additional work pending in the event loop. The process.exitCode property can be set to tell the process which exit code to use when the process exits gracefully.